assets/app/scripts/directives/oscFileInput.js

```diff
@@ -31,6 +49,67 @@ angular.module('openshiftConsole')
           };
           reader.readAsText(file);
         });
+
+        // Specify event handler for the whole page drag & drop actions
+        $(document).on('dragover', function(e) {
+          preventAndStop(e);
+          if (isDisabled) {
+            return false;
+          }
```

Comment:
I think I'd rather scope.$watch('disabled', ...) and add and remove the listeners instead of checking disabled inside the listener.